jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/QuotedCSVParser.java

```diff
@@ -155,23 +160,28 @@ else if (state == State.PARAM_VALUE)
                         switch (state)
                         {
                             case VALUE:
-                                parsedValue(buffer);
+                                valueLength = buffer.length();
+                                if (valueLength > 0 || paramsOnly)
+                                    parsedValue(buffer);
                                 break;
                             case PARAM_NAME:
                             case PARAM_VALUE:
-                                parsedParam(buffer, valueLength, paramName, paramValue);
+                                if (valueLength > 0 || paramsOnly)
+                                    parsedParam(buffer, valueLength, paramName, paramValue);
                                 break;
                             default:
                                 throw new IllegalStateException(state.toString());
                         }
 
-                        parsedValueAndParams(buffer, valueLength);
+                        if (valueLength > 0 || paramsOnly)
+                            parsedValueAndParams(buffer);
```

Comment:
I'm surprised this didn't break QuotedCSVTest (I did this same change before, and it broke expectations there)